### ISSUE TITLE
TypeError in transforms notebook

### ISSUE THREAD

**Issue by fepegar** [Project-MONAI/MONAI#851]
Hi all,

I'm trying to run the `integrate_3rd_party_transforms` notebook on Google Colab, but it's not working. First, there's no line to install MONAI. Second, once installed, I get this error when trying to run the last cell:

```python-traceback
TypeError                                 Traceback (most recent call last)
<ipython-input-15-dfd10a517436> in <module>()
      2 check_loader = monai.data.DataLoader(check_ds, batch_size=1)
      3 check_data = monai.utils.misc.first(check_loader)
----> 4 image, label = (check_data['image'][0][0], check_data['label'][0][0])
      5 print(f"image shape: {image.shape}, label shape: {label.shape}")
      6 # plot the slice [:, :, 80]

TypeError: 'NoneType' object is not subscriptable
```

**Comment by fepegar** [comment]
By the way, (most) TorchIO transforms now take dictionaries as input:

```python
import torch
import torchio
t = torch.rand(1, 28, 14, 14)
data = {'a': 1, 'image': t, 'image2': t*2}
result = torchio.Resample(0.5, keys=['image'])(data)
```

I hope this will make integration with MONAI easier.

**Comment by Nic-Ma** [comment]
Hi @fepegar ,

Thanks very much for your effort to help verify this notebook.
We didn't test this notebook in Colab so far.
Could you please help double confirm whether the Decathlon dataset was loaded correctly?
And whether version of TorchIO can support dict now? I can update the notebook for this part later.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @fepegar ,

I created a PR https://github.com/Project-MONAI/MONAI/pull/860 to update the TorchIO usage in the tutorial.
Thanks.

**Comment by fepegar** [comment]
Hi @Nic-Ma,

> We didn't test this notebook in Colab so far. Could you please help double confirm whether the Decathlon dataset was loaded correctly?

Oh, sorry. I see now that the path to the data was local. I thought the notebook would be self-contained. I think it would be nice that

1. It was indeed self-contained and one can just press "Run all". MONAI can download the needed decathlon dataset, is that right?
2. A warning was raised if the input of a transform was empty. Have you considered this?

> And whether version of TorchIO can support dict now? I can update the notebook for this part later.

It does, it was added in https://github.com/fepegar/torchio/commit/19ff6e7568ab2c62e579fd0728571760d3e381c4 and it's available in `v0.17.21` (you can try running the code I shared in https://github.com/Project-MONAI/MONAI/issues/851#issuecomment-669139528). I haven't added support in the transforms mentioned in the commit, message, but it should be fine for most of them. 

If the input is a `dict`, a `keys` kwarg must be passed. The values of those keys are expected to be 4D tensors with shape (C, D, H, W). I can adapt them further if that would be helpful.

**Comment by Nic-Ma** [comment]
Hi @fepegar ,

We may try to move all notebooks to Colab later, and update to use DecathlonDataset.
Because we want to do a new refactoring iteration for datasets to support cross-validation soon, so I haven't changed notebooks.

Thanks.
